svAssistant.py

```diff
@@ -22,88 +23,42 @@
 
-crops_growth_values = {
-# 'grow_time' is the number of days it takes for the crop to completely grow.
-# 'produce_time' is the number of days it takes to reproduce yield post-growth.
-# 'harvests/crop' is the number of times the crop can be harvested per season.
-# '999' for 'harvests/crop' means infinite.
-# 'yield' is the number of crops you recieve per harvest.
-"Cauliflower": {"grow_time": 12, "harvests/crop": 1, "yield": 1},
-"Garlic": {"grow_time": 4, "harvests/crop": 1, "yield": 1},
-"Green Bean": {"grow_time": 10, "produce_time": 3, "harvests/crop": 999, "yield": 1},
-"Kale": {"grow_time": 6, "harvests/crop": 1, "yield": 1},
-"Parsnip": {"grow_time": 4, "harvests/crop": 1, "yield": 1},
-"Potato": {"grow_time": 6, "harvests/crop": 1, "yield": 1.25},
-"Rhubarb": {"grow_time": 13, "harvests/crop": 1, "yield": 1},
-"Strawberry": {"grow_time": 8, "produce_time": 4, "harvests/crop": 999, "yield": 1},
-"Blue Jazz": {"grow_time": 7, "harvests/crop": 1, "yield": 1},
-"Tulip": {"grow_time": 6, "harvests/crop": 1, "yield": 1},
-"Blueberry": {"grow_time": 13, "produce_time": 4, "harvests/crop": 999, "yield": 3},
-"Corn": {"grow_time": 14, "produce_time": 4, "harvests/crop": 999, "yield": 1},
-"Hops": {"grow_time": 11, "produce_time": 1, "harvests/crop": 999, "yield": 1},
-"Hot Pepper": {"grow_time": 5, "produce_time": 3, "harvests/crop": 999, "yield": 1},
-"Melon": {"grow_time": 12, "harvests/crop": 1, "yield": 1},
-"Radish": {"grow_time": 6, "harvests/crop": 1, "yield": 1},
-"Red Cabbage": {"grow_time": 9, "harvests/crop": 1, "yield": 1},
-"Starfruit": {"grow_time": 13, "harvests/crop": 1, "yield": 1},
-"Tomato": {"grow_time": 11, "produce_time": 4, "harvests/crop": 5, "yield": 1},
-"Wheat": {"grow_time": 4, "harvests/crop": 1, "yield": 1},
-"Poppy": {"grow_time": 7, "harvests/crop": 1, "yield": 1},
-"Summer Spangle": {"grow_time": 8, "harvests/crop": 1, "yield": 1},
-"Amaranth": {"grow_time": 7, "harvests/crop": 3, "yield": 1},
-"Artichoke": {"grow_time": 8, "harvests/crop": 1, "yield": 1},
-"Beet": {"grow_time": 6, "harvests/crop": 1, "yield": 1},
-"Bok Choy": {"grow_time": 4, "harvests/crop": 1, "yield": 1},
-"Cranberries": {"grow_time": 7, "produce_time": 5, "harvests/crop": 5, "yield": 2},
-"Eggplant": {"grow_time": 5, "produce_time": 5, "harvests/crop": 999, "yield": 1},
-"Grape": {"grow_time": 10, "produce_time": 3, "harvests/crop": 999, "yield": 1},
-"Pumpkin": {"grow_time": 13, "harvests/crop": 1, "yield": 1},
-"Yam": {"grow_time": 10, "harvests/crop": 1, "yield": 1},
-"Fairy Rose": {"grow_time": 12, "harvests/crop": 1, "yield": 1},
-"Sunflower": {"grow_time": 8, "harvests/crop": 1, "yield": 1}
-}
+# Creates a dictionary containing the buy and sell price of crops. Retrieves
+# values from crops_store_values.csv
+crops_store_values = {}
+with open('crops_store_values.csv') as crop_values:
+    read_csv = csv.reader(crop_values, delimiter=',')
+
+    for row in read_csv:
+        if len(row) == 0 or row[0][0] == "#":
```

Comment:
@mtlynch Don't entirely agree here.  Various csv parsers do support ignoring lines that start with some form of comment character. Often '#'.  Since he is defining his own parser I think it's not a bad way to comment the file.  However if he switches to DictReader then it may no longer be an option.

See the pandas.read_csv for handling large datasets. http://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_csv.html